Source/DafnyCore/Substituter.cs

```diff
@@ -930,7 +979,73 @@ public class Substituter {
           }
         }
       }
+
       return attrs;
     }
+
+    /// <summary>
+    /// Substitution in a comprehension expression. 
+    /// Parameter 'forceSubstituteOfBoundVars' should be set to false if and only if
+    /// the expression is a binding guard, in which case a bound variable is introduced.
+    /// Such a variable must not be substituted. 
+    /// </summary>
+    protected Expression SubstituteComprehensionExpr(ComprehensionExpr expr, bool forceSubstituteOfBoundVars) {
+
+      Expression newExpr = null;
```

Comment:
Could this be a `ComprehensionExpr` instead of an `Expr`?

Reply:
Not sure. ```SplitQuantifierExpression``` returns an Expression, and not a ComprehensionExpression.